programs/bench.c

```diff
@@ -490,12 +501,27 @@ static int BMK_benchMem(const void* srcBuffer, size_t srcSize,
     free(blockTable);
     free(compressedBuffer);
     free(resultBuffer);
+    return results;
+}
+
+static void BMK_benchMemCtxless(const void* srcBuffer, size_t srcSize,
+                        const size_t* fileSizes, unsigned nbFiles, 
+                        int cLevel, const ZSTD_compressionParameters* const comprParams,
+                        const void* dictBuffer, size_t dictBufferSize,
+                         int displayLevel, const char* displayName) 
+{
+    ZSTD_CCtx* ctx = ZSTD_createCCtx();
+    ZSTD_DCtx* dctx = ZSTD_createDCtx();
```

Comment:
Important : check that `ctx` and `dctx` are both non NULL !
Memory allocation can always fail !

Note : in this case, it's an active check, with an associated "error" behavior defined.
Not to be confused with an `assert()`, which describes a situation which should never happen.